### DISCUSSION TITLE
GitHub Classroom and Codespaces

### DISCUSSION THREAD

**Discussion by sierrajulietromeo** [community/community#121826] [GitHub Education] [answered]
### Hi everyone,

We've successfully run a few modules with programming assignments with groups sized 20-25 using GitHub Classroom with Codespaces activated.  Has all worked wonderfully.

I'm verified, so we get the following free monthly allowance as described [here](https://docs.github.com/en/education/manage-coursework-with-github-classroom/integrate-github-classroom-with-an-ide/using-github-codespaces-with-github-classroom
). 

I'm interested in this specific part: 

![image](https://github.com/community/community/assets/30177706/e899cc46-f29d-4ade-aac1-11e6beca4f47)

We want to roll out to larger groups.   I know (as above) currently that it's in 'beta', and if I check our organisation usage report, it shows 0, even though we've had students using Codespaces.  (I assume because you aren't charging us for it).

What I'd like to know is, 

1) according to above, it's sufficient for 50 students above completing 5 assignments.  How long are you assuming that an assignment takes to complete?

2) Also, what is pricing going to be like when the Beta finishes, and what notice will we get.   Knowing the answers to these questions will help us better plan for groups > 50 with more assignments.

3) Our students have often had more than one Codespace inactive/stored.  (Usually somewhere in the region of 3 to 10).  What are the likely costs for this?

Thanks

**Comment by RyanHecht** [comment, marked as answer]
Hi @sierrajulietromeo! I'm glad to hear you're finding value in Codespaces in your classroom! I'm really excited about the opportunity it presents for teachers.

> according to above, it's sufficient for 50 students above completing 5 assignments. How long are you assuming that an assignment takes to complete?

We're estimating 5 hours per assignment to calculate this number!

> Also, what is pricing going to be like when the Beta finishes, and what notice will we get. Knowing the answers to these questions will help us better plan for groups > 50 with more assignments.

I'm sorry to say that we don't have an estimate for any of this just yet! I'm saving this discussion to update you when we have more clarity, though!

> Our students have often had more than one Codespace inactive/stored. (Usually somewhere in the region of 3 to 10). What are the likely costs for this?

[Codespaces compute usage is billed for the duration for which a codespace is active](https://docs.github.com/en/codespaces/setting-your-user-preferences/setting-your-timeout-period-for-github-codespaces#:~:text=Codespaces%20compute%20usage%20is%20billed%20for%20the%20duration%20for%20which%20a%20codespace%20is%20active.), so there shouldn't be any costs for inactive Codespaces!

**Reply by sierrajulietromeo** [reply]
Thanks @RyanHecht for your reply. 5 hours per assignment is a reasonable number for us.
We've been really pleased with how Codespaces has worked for us, and it's allowed us to dodge so many issues regarding local installations, permissions etc. Really great.   Though plenty have been opening on their local VS Code via the option to do so, and that's also been flawless.

We'll certainly continue to use it going forward and we look forward to continuing use with GitHub classroom.
Look forward to hearing pricing details when you know more, but we're certainly happy enough whilst it's in 'free' public beta.

I'll mark this as answered. Cheers!
